### PR TITLE
#5204: Fix Intermittent smoketest failures due to parallel testing

### PR DESCRIPTION
**Pull request by kglazko**
*Thank you for submitting a pull request, your contributions are greatly appreciated!*

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Please make sure you've run the test scheme and all the tests pass. If your changes affect existing tests please make sure to update the tests as well.

### CODE REVIEW

File: XCUITests/TopTabsTest.swift [hunk 1]
```diff
@@ -24,6 +24,7 @@ class TopTabsTest: BaseTestCase {
         waitUntilPageLoad()
         waitForTabsButton()
         // The tabs counter shows the correct number
+        waitForExistence(app.buttons["Show Tabs"])
```
Comment (isabelrios) [inline XCUITests/TopTabsTest.swift]: This does not seem to work on Bitrise, this test is always failing now...I have been running your branch there (see [logs](https://app.bitrise.io/build/326cc9a4bef4f79b#?tab=log)) 

What you could do is to increase the waiting time in the `waitForTabsButton()`which is doing the wait for that screen to be shown

**Comment by isabelrios** [comment]
I know this is WIP :) but was taking a look at it to try to help fixing the intermittent failures. 
One suggestion would be to use 'waitForExistence(...., timeout: x) where x is a number of secs, that could help to wait for an element...
Also, you will see on bitrise your branch, you can rebuild it and see the results once you make changes here.
Thanks!

**Comment by isabelrios** [comment]
I would say there are less intermittents...Still this test `testRearrangeTabsTabTray()`fails very often and there is no solution with waits as far as I now...I think we can disable it from the smoketest and enable in the regular test suite... what do you think @kglazko?

File: XCUITests/TopTabsTest.swift [hunk 2]
```diff
@@ -24,7 +24,7 @@ class TopTabsTest: BaseTestCase {
         waitUntilPageLoad()
         waitForTabsButton()
         // The tabs counter shows the correct number
-        waitForExistence(app.buttons["Show Tabs"])
+        waitForExistence(app.buttons["Show Tabs"], timeout: 10)
```
Comment (isabelrios) [inline XCUITests/TopTabsTest.swift]: This `waitForExistence is redundant`, this is already checked in the previous statement, please remove from here...

**Comment by kglazko** [comment]
Trying one last thing, if not, will disable for Smoketests RearrangeTabs test

File: XCUITests/TopTabsTest.swift [hunk 3]
```diff
@@ -190,27 +190,31 @@ class TopTabsTest: BaseTestCase {
     func testLongTapTabCounter() {
         if !iPad() {
             // Long tap on Tab Counter should show the correct options
-            waitForExistence(app.buttons["Show Tabs"], timeout: 5)
+            waitForExistence(app.buttons["Show Tabs"], timeout: 10)
             app.buttons["Show Tabs"].press(forDuration: 1)
             waitForExistence(app.cells["quick_action_new_tab"])
             XCTAssertTrue(app.cells["quick_action_new_tab"].exists)
+            waitForExistence(app.cells["tab_close"])
```
Comment (isabelrios) [inline XCUITests/TopTabsTest.swift]: No need to add this `waitForExists`, it is waiting for an option shown in the same menu as the previous option and there is already a wait for it

File: XCUITests/TopTabsTest.swift [hunk 4]
```diff
@@ -190,27 +190,31 @@ class TopTabsTest: BaseTestCase {
...
             checkNumberOfTabsExpectedToBeOpen(expectedNumberOfTabsOpen: 2)
+            waitForExistence(app.collectionViews.cells["Home"])
             app.collectionViews.cells["Home"].firstMatch.tap()
 
             // Close tab
             navigator.nowAt(HomePanelsScreen)
             waitForExistence(app.buttons["Show Tabs"])
             app.buttons["Show Tabs"].press(forDuration: 1)
             waitForExistence(app.cells["quick_action_new_tab"])
+            waitForExistence(app.cells["tab_close"])
```
Comment (isabelrios) [inline XCUITests/TopTabsTest.swift]: Same here, no need for this wait

**Comment by isabelrios** [comment]
Hey @kglazko this is looking much better! I got two green builds in a row! :) 
I left a few comments, and also, there are files that should not be included in the PR. Apart from that, I think we could try with these changes and improve from there having more results...

**Comment by isabelrios** [comment]
@kglazko there are still files changed that should not be merged: ` Client/Assets/MainFrameAtDocumentEnd.js`, ` Client/Assets/MainFrameAtDocumentStart.js`and ` package-lock.json`
The test `testRearrangeTabsTabTray()` is a little bit unstable when running the tests this way, please disable it from smoke test and enable en regular test.
Please, let's do these changes and land the PR asap to get back green builds.

**Comment by isabelrios** [comment]
@kglazko great work here! it looks good to me, if that is finished on your side too you can remove the WIP and merge it! thanks!

**Pull request merged**